### PR TITLE
#370_cant_set_minimum_log_scale_in_cuts

### LINKED ISSUE #370: Can't set cut axis minimum in logarithmic scale
If the y axis of a cut is set to logarithmic, the minimum cannot be changed.

### PR DESCRIPTION
**Pull request by aybamidele**
Description of work.
Removed bug that prevents user from modifying the lower bound in the log scale in interactive cuts
**To test:**

Perform a cut in mslice, change the scale to logarithmic. The lower bound of the range can be modified

Fixes #370 .

### CODE REVIEW

**Comment by aybamidele** [comment]
@martyngigg I get these runtime errors when I do this.

> C:\MantidInstall\bin\lib\site-packages\matplotlib\scale.py:341: RuntimeWarning: invalid value encountered in sign
  sign = np.sign(a)
C:\MantidInstall\bin\lib\site-packages\numpy\ma\core.py:2023: RuntimeWarning: invalid value encountered in greater_equal
  condition = (xf >= v1) & (xf <= v2)
C:\MantidInstall\bin\lib\site-packages\numpy\ma\core.py:2023: RuntimeWarning: invalid value encountered in less_equal
  condition = (xf >= v1) & (xf <= v2)
C:\MantidInstall\bin\lib\site-packages\matplotlib\scale.py:348: RuntimeWarning: invalid value encountered in absolute
  ma.log(np.abs(masked) / self.linthresh) / self._log_base)
C:\MantidInstall\bin\lib\site-packages\numpy\ma\core.py:790: RuntimeWarning: invalid value encountered in greater_equal
  return umath.absolute(a) * self.tolerance >= umath.absolute(b)
C:\MantidInstall\bin\lib\site-packages\numpy\ma\core.py:802: RuntimeWarning: invalid value encountered in less_equal
  return umath.less_equal(x, self.critical_value)

**Comment by aybamidele** [comment]
@martyngigg, I get runtime errors when I try to do this. 

> C:\MantidInstall\bin\lib\site-packages\matplotlib\scale.py:341: RuntimeWarning: invalid value encountered in sign
  sign = np.sign(a)
C:\MantidInstall\bin\lib\site-packages\numpy\ma\core.py:2023: RuntimeWarning: invalid value encountered in greater_equal
  condition = (xf >= v1) & (xf <= v2)
C:\MantidInstall\bin\lib\site-packages\numpy\ma\core.py:2023: RuntimeWarning: invalid value encountered in less_equal
  condition = (xf >= v1) & (xf <= v2)
C:\MantidInstall\bin\lib\site-packages\matplotlib\scale.py:348: RuntimeWarning: invalid value encountered in absolute
  ma.log(np.abs(masked) / self.linthresh) / self._log_base)
C:\MantidInstall\bin\lib\site-packages\numpy\ma\core.py:790: RuntimeWarning: invalid value encountered in greater_equal
  return umath.absolute(a) * self.tolerance >= umath.absolute(b)
C:\MantidInstall\bin\lib\site-packages\numpy\ma\core.py:802: RuntimeWarning: invalid value encountered in less_equal
  return umath.less_equal(x, self.critical_value)

**Comment by martyngigg** [comment]
@aybamidele What values are you using to test it?

**Comment by martyngigg** [comment]
Is that the whole traceback?

**Comment by aybamidele** [comment]
I'm using values greater than 0. This is the whole traceback:

> C:\MantidInstall\bin\lib\site-packages\numpy\ma\core.py:3900: UserWarning: Warning: converting a masked element to nan.
  warnings.warn("Warning: converting a masked element to nan.")
C:\MantidInstall\bin\lib\site-packages\matplotlib\scale.py:341: RuntimeWarning: invalid value encountered in sign
  sign = np.sign(a)
C:\MantidInstall\bin\lib\site-packages\numpy\ma\core.py:2023: RuntimeWarning: invalid value encountered in greater_equal
  condition = (xf >= v1) & (xf <= v2)
C:\MantidInstall\bin\lib\site-packages\numpy\ma\core.py:2023: RuntimeWarning: invalid value encountered in less_equal
  condition = (xf >= v1) & (xf <= v2)
C:\MantidInstall\bin\lib\site-packages\matplotlib\scale.py:348: RuntimeWarning: invalid value encountered in absolute
  ma.log(np.abs(masked) / self.linthresh) / self._log_base)
C:\MantidInstall\bin\lib\site-packages\numpy\ma\core.py:790: RuntimeWarning: invalid value encountered in greater_equal
  return umath.absolute(a) * self.tolerance >= umath.absolute(b)
C:\MantidInstall\bin\lib\site-packages\numpy\ma\core.py:802: RuntimeWarning: invalid value encountered in less_equal
  return umath.less_equal(x, self.critical_value)

**Comment by aybamidele** [comment]
The errors only show up after the first scaling. It seems to work as expected afterwards.

File: mslice/plotting/plot_window/cut_plot.py [hunk 1]
```diff
@@ -100,16 +100,14 @@ def update_legend(self, line_data=None):
     def change_axis_scale(self, xy_config):
         current_axis = self._canvas.figure.gca()
         if xy_config['x_log']:
-            xdata = [ll.get_xdata() for ll in current_axis.get_lines()]
-            xmin = get_min(xdata, absolute_minimum=0.)
+            xmin = xy_config['x_range'][0]
```
Comment (martyngigg) [inline mslice/plotting/plot_window/cut_plot.py]: `symlog` should be able to handle a negative scale <= 0 so what I think we need to do here is set the min/max scale to the true values specified by the user in `xy_config` but then use `get_min(xdata, absolute_minimum=0.)` to calculate the number that goes into `linthreshx` and not use `xmin` there.

This should gives us a log scale that is linear below the minimum data values.

File: mslice/plotting/plot_window/cut_plot.py [hunk 2]
```diff
@@ -100,16 +100,14 @@ def update_legend(self, line_data=None):
...
             current_axis.set_xscale('symlog', linthreshx=pow(10, np.floor(np.log10(xmin))))
             if xmin > 0:
                 xy_config['x_range'] = (xmin, xy_config['x_range'][1])
         else:
             current_axis.set_xscale('linear')
         if xy_config['y_log']:
```
Comment (martyngigg) [inline mslice/plotting/plot_window/cut_plot.py]: Same comment as above for `y`

**Comment by martyngigg** [comment]
@aybamidele Minor point. Could you make sure you set a good title for the pull request (you can edit it with the Edit button at the top) and also assign a milestone.

**Pull request closed without merging**